usr/lib/pkcs11/cca_stdll/cca_specific.c

```diff
@@ -2858,6 +2859,12 @@ static CK_RV rsa_import_privkey_crt(TEMPLATE * priv_tmpl)
         return rc;
     }
 
+    OPENSSL_cleanse(p_prime->pValue, p_prime->ulValueLen);
+    OPENSSL_cleanse(q_prime->pValue, q_prime->ulValueLen);
+    OPENSSL_cleanse(dmp1->pValue, dmp1->ulValueLen);
+    OPENSSL_cleanse(dmq1->pValue, dmq1->ulValueLen);
+    OPENSSL_cleanse(iqmp->pValue, iqmp->ulValueLen);
```

Comment:
Also cleanse sensitive parts of key_value_structure. This function needs a goto label exit point:
```
    rv = CKR_OK
err:
    cleanse sesnsitive parts of key_value_struct
    return rv
```
All return points after the first sensitive value has been copied to key_value_struct should be mapped to that exit point.

Reply:
Doesnt store priv_tmpl also the private exponent which has to be erased ?

Reply:
good catch. Fixed (both comments)